acceptance/assertions/output.go

```diff
@@ -0,0 +1,276 @@
+// +build acceptance
+
+package assertions
+
+import (
+	"fmt"
+	"regexp"
+	"testing"
+
+	h "github.com/buildpacks/pack/testhelpers"
+)
+
+type OutputAssertionManager struct {
+	testObject *testing.T
+	assert     h.AssertionManager
+	output     string
+}
+
+func NewOutputAssertionManager(t *testing.T, output string) OutputAssertionManager {
+	return OutputAssertionManager{
+		testObject: t,
+		assert:     h.NewAssertionManager(t),
+		output:     output,
+	}
+}
+
+func (o OutputAssertionManager) ReportsSuccessfulImageBuild(name string) {
+	o.testObject.Helper()
+
+	o.assert.ContainsF(o.output, "Successfully built image '%s'", name)
+}
+
+func (o OutputAssertionManager) ReportsSuccessfulRebase(name string) {
+	o.testObject.Helper()
+
+	o.assert.ContainsF(o.output, "Successfully rebased image '%s'", name)
+}
+
+func (o OutputAssertionManager) ReportsUsingBuildCacheVolume() {
+	o.testObject.Helper()
+
+	o.testObject.Log("uses a build cache volume")
+	o.assert.Contains(o.output, "Using build cache volume")
+}
+
+func (o OutputAssertionManager) ReportsSelectingRunImageMirror(mirror string) {
+	o.testObject.Helper()
+
+	o.testObject.Log("selects expected run image mirror")
+	o.assert.ContainsF(o.output, "Selected run image mirror '%s'", mirror)
+}
+
+func (o OutputAssertionManager) ReportsSelectingRunImageMirrorFromLocalConfig(mirror string) {
+	o.testObject.Helper()
+
+	o.testObject.Log("local run-image mirror is selected")
+	o.assert.ContainsF(o.output, "Selected run image mirror '%s' from local config", mirror)
+}
+
+func (o OutputAssertionManager) ReportsRestoresCachedLayer(layer string) {
+	o.testObject.Helper()
+	o.testObject.Log("restores the cache")
+
+	o.assert.MatchesAll(
+		o.output,
+		regexp.MustCompile(fmt.Sprintf(`(?i)Restoring data for "%s" from cache`, layer)),
+		regexp.MustCompile(fmt.Sprintf(`(?i)Restoring metadata for "%s" from app image`, layer)),
+	)
+}
+
+func (o OutputAssertionManager) ReportsExporterReusingUnchangedLayer(layer string) {
+	o.testObject.Helper()
+	o.testObject.Log("exporter reuses unchanged layers")
+
+	o.assert.Matches(o.output, regexp.MustCompile(fmt.Sprintf(`(?i)Reusing layer '%s'`, layer)))
+}
+
+func (o OutputAssertionManager) ReportsCacheReuse(layer string) {
```

Comment:
Related to the `OutputAssertionManager` being "overloaded".

Another potential improvement here may be to create a separation for the parties producing the output. For instance this output is produced by the lifecycle and not pack so something like `LifecycleOutputAssertionManager` would further encapsulate and clarify responsibilities.